components/sml.rst

```diff
@@ -101,6 +108,21 @@ Text Sensor
 - **format** (*Optional*, string): Override the automatic interpretation of the transmitted binary data value. Possible values (`int`, `uint`, `bool`, `hex`, `text`).
 - All other options from :ref:`Text Sensor <config-text_sensor>`.
 
+Automations:
+------------
+
+- **on_data** (*Optional*, :ref:`Automation <automation>`): An automation to perform when a
+  valid SML message is received. See :ref:`sml-on-data`.
```

Comment:
It is still called when not valid
```suggestion
- **on_data** (*Optional*, :ref:`Automation <automation>`): An automation to perform when a
  SML message is received. See :ref:`sml-on-data`.
```

Reply:
True. I changed afterwards by adding the "valid" variable so that consumers can decide what to do with invalid sml data.